pkg/readiness/object_tracker.go

```diff
@@ -249,7 +281,7 @@ func (t *objectTracker) Satisfied() bool {
 		}
 		delete(t.seen, k)
 		delete(t.expect, k)
-		t.satisfied[k] = struct{}{}
+		t.satisfied[k] = mutatedObjData(t.mutators...)
```

Comment:
We probably want to leave this as an empty struct to avoid unnecessary memory usage.

Reply:
I actually can't do that anymore, because the value is now of type `objData` (since it has the `retries` field in it).  Is there a different implementation you'd suggest that dodges this?

Passing in a raw `objData` would save a call to a mutation function, but it's O(1).

Reply:
You could just have the different arrays have different types. That being said, I'm not sure the extra memory savings is worth the extra complexity.

@shomron any opinions on uniformity vs. lower memory usage here? IIRC this gets GCd after bootstrapping anyway.

Reply:
> You could just have the different arrays have different types

I made the same suggestion for other reasons (see my review). I'm not worried about the memory usage (and yes we nuke these maps after bootstrapping) - but more about how we can reason about the code.